src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,46 @@
+package duke.command;
+
+import java.util.ArrayList;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+/**
+ * Command: Finds a list of task that matches the query
+ */
+public class FindCommand extends Command {
+    private String query;
+
+    /**
+     * Creates Find Command
+     * @param query string from user
+     */
+    public FindCommand(String query) {
+        this.query = query;
+    }
+
+    /**
+     * Executes the command
+     *
+     * @param tasks   TaskList object to get and set the list
+     * @param ui      object to reply to user after the command has executed
+     * @param storage object required when command writes to file
+     * @throws DukeException
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        ArrayList<Task> result = tasks.search(query);
```

Comment:
Should the variable name result be changed to results as it contains a collection of Tasks?